### PR TITLE
Convert README to markdown

### PR DESCRIPTION
**Pull request by Marcono1234**
Notable changes:
- Link to BUILDING, LICENSE and NOTICE
- Move "Contact" section above "Cryptographic Software Notice" because it is likely more important for most users
- Use HTTPS links

Preview can be seen [here](https://github.com/Marcono1234/httpcomponents-client/blob/marcono1234/readme/README.md).

Maybe it would also be good to link to the javadoc? However based on the URL https://hc.apache.org/httpcomponents-client-5.0.x/httpclient5/apidocs/ seems to be only for version 5.0.x, therefore this link could easily become outdated once 5.1 or higher is released. Maybe it would be better to use a [javadoc.io](https://javadoc.io/doc/org.apache.httpcomponents.client5/httpclient5/latest/index.html) link instead (though that shows the javadoc for the beta version by default)?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,30 +3,44 @@ Apache HttpComponents Client
...
 ------------
 
 HttpClient main module requires Java 7 compatible runtime and
 depends on the following external libraries:
 
-* Apache HttpComponents HttpCore
-* Apache Logging Log4j2 API
-* Apache Commons Codec
+* [Apache HttpComponents HttpCore](https://github.com/apache/httpcomponents-core)
+* [Apache Logging Log4j2 API](https://github.com/apache/logging-log4j2)
```
Comment (michael-o) [inline README.md:22]: I wonder where this comes from. Seems to be old. Can be dropped.

**Review by michael-o — CHANGES_REQUESTED** [review]

**Review by ok2c — CHANGES_REQUESTED** [review]
@Marcono1234  We need to keep README.txt because it gets shipped with release dist packages. README.md must have a ASLv2 header as in HttpCore. Looks good otherwise.

**Comment by michael-o** [comment]
> 
> 
> @Marcono1234 We need to keep README.txt because it gets shipped with release dist packages. README.md must have a ASLv2 header as in HttpCore. Looks good otherwise.

Where does the text requirement come from? Markdown is so simple that most people can render it in their brain directly.

File: README.md (hunk 1 above)
Comment (ok2c) [reply README.md:22]: @michael-o It is not old but wrong. The correct dependencies are
* Apache HttpComponents HttpCore
* SLF4J
* Apache Commons Codec

Comment (michael-o) [reply README.md:22]: These are correct now.

**Comment by ok2c** [comment]
> Where does the text requirement come from? Markdown is so simple that most people can render it in their brain directly.

It is not a requirement but rather a tradition. If you think we should remove README.txt in favor of README.md I suppose I can live with that but we should be consistent and do it HttpCore as well.

**Comment by Marcono1234** [comment]
@ok2c which license header should I use? HttpCore includes "Copyright (C) 2016 Gary Gregory", should I maybe instead use the one from pom.xml?

**Comment by ok2c** [comment]
> @ok2c which license header should I use? HttpCore includes "Copyright (C) 2016 Gary Gregory", should I maybe instead use the one from pom.xml?

That looks odd. @garydgregory, any idea why your name is README.md copyright header? Was that intentional?

@Marcono1234  Please use the one from pom.xml

**Comment by Marcono1234** [comment]
Force-pushed to include the license header in commit "Convert README to markdown".

**Comment by garydgregory** [comment]
> > @ok2c which license header should I use? HttpCore includes "Copyright (C) 2016 Gary Gregory", should I maybe instead use the one from pom.xml?
> 
> That looks odd. @garydgregory, any idea why your name is README.md copyright header? Was that intentional?
> 
> @Marcono1234 Please use the one from pom.xml

"Copyright (C) 2016 Gary Gregory"
That must be a mistake by me when adding a new file.

**Comment by garydgregory** [comment]
FWIW, we use readme in markdown in all or most Apache Commons components.

**Comment by michael-o** [comment]
> 
> 
> FWIW, we use readme in markdown in all or most Apache Commons components.

I'd to the same here.

**Review by ok2c — APPROVED** [review]

**Comment by ok2c** [comment]
@michael-o @garydgregory Fine by me. Let us remove README.txt from both core and client.

**Pull request merged**